### PR TITLE
Update beanValidationCore.mustache

### PR DESCRIPTION
**Pull request by matheustf**
Update to use x-pattern-message for message customization

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by matheustf** [comment]
Update to use x-pattern-message for message customization

**Comment by rodrigoma3** [comment]
@cachescrubber @welshm @MelleD @borsch @Zomzog @atextor @manedev79 @javisst @banlevente 
Can you merge pull request please?

**Comment by rodrigoma3** [comment]
@wing328 ping

**Comment by welshm** [comment]
It would be good to add tests for this if possible - I don't see any issues with teh change though, but might be good to document the usage of `x-pattern-message` - https://openapi-generator.tech/docs/generators/spring#supported-vendor-extensions

**Comment by matheustf** [comment]
@cachescrubber @welshm @MelleD @borsch @Zomzog @atextor @manedev79 @javisst @banlevente
We update with the documentation. Can you merge pull request please?

File: docs/generators/spring.md [hunk 1]
```diff
@@ -113,6 +113,7 @@ These options may be applied as additional-properties (cli) or configOptions (pl
 |x-field-extra-annotation|List of custom annotations to be added to property|FIELD|null
 |x-spring-paginated|Add org.springframework.data.domain.Pageable to controller method. Can be used to handle page & size query parameters|OPERATION|false
 |x-version-param|Marker property that tells that this parameter would be used for endpoint versioning. Applicable for headers & query params. true/false|OPERATION_PARAMETER|null
+|x-pattern-message| Add this property whenever you need to customize the invalidation error message for the regex pattern of a variable|FIELD|
```
Comment (borsch) [inline docs/generators/spring.md:116]: we don't add this manually. Add this to https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/VendorExtension.java + add it as supported extension to https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java#L1241


once all this done you would need to re-generated docs using
`mvn clean && ./bin/utils/export_docs_generators.sh`

**Review by borsch — CHANGES_REQUESTED** [review]
Please update docs accordingly

**Comment by rodrigoma3** [comment]
@borsch 
done

**Review by borsch — APPROVED** [review]

**Pull request merged**